app/database/models/personal_background.py

```diff
@@ -0,0 +1,152 @@
+    """Defines attributes for user's personal background.
+
+    Attributes:
+    user_id: An integer for storing the user's id.
+    gender: A string for storing the user's gender.
+    age: A string for storing the user's age.
+    ethnicity: A string for storing the user's wthnicity.
+    sexual_orientation: A string for storing the user's sexual orientation.
+    religion: A string for storing the user's religion.
+    physical_ability: A string for storing the user's physical ability.
+    mental_ability: A string for storing the user's mental ability.
+    socio_economic: A string for storing the user's socio economic level.
+    highest_education: A string for storing the user's highest education level.
+    years_of_experience: A string for storing the user's length of expeprience in the It related area.
+    others: A JSON data type for storing users descriptions of 'other' fields.
+    is_public: A boolean indicating if user has agreed to display their personal background information publicly to other members.
+    """
+
+    # Specifying database table used for PersonalBackgroundModel
+    __tablename__ = "personal_backgrounds"
+    __table_args__ = {"extend_existing": True}
+
+    id = db.Column(db.Integer, primary_key=True)
+
+    # User's personal background data
+    user_id = db.Column(
+        db.Integer,
+        db.ForeignKey("users.id", ondelete="CASCADE"),
+        nullable=False,
+        unique=True,
+    )
+    gender = db.Column(db.Enum(Gender))
+    age = db.Column(db.Enum(Age))
+    ethnicity = db.Column(db.Enum(Ethnicity))
+    sexual_orientation = db.Column(db.Enum(SexualOrientation))
+    religion = db.Column(db.Enum(Religion))
+    physical_ability = db.Column(db.Enum(PhysicalAbility))
+    mental_ability = db.Column(db.Enum(MentalAbility))
+    socio_economic = db.Column(db.Enum(SocioEconomic))
+    highest_education = db.Column(db.Enum(HighestEducation))
+    years_of_experience = db.Column(db.Enum(YearsOfExperience))
+    others = db.Column(JsonCustomType)
+    is_public = db.Column(db.Boolean)
+
+    def __init__(
+        self,
+        user_id,
+        gender,
+        age,
+        ethnicity,
+        sexual_orientation,
+        religion,
+        physical_ability,
+        mental_ability,
+        socio_economic,
+        highest_education,
+        years_of_experience,
+    ):
+        """Initialises PersonalBackgroundModel class."""
+        ## required fields
+        self.user_id = user_id
+        self.gender = gender
+        self.age = age
+        self.ethnicity = ethnicity
+        self.sexual_orientation = sexual_orientation
+        self.religion = religion
+        self.physical_ability = physical_ability
+        self.mental_ability = mental_ability
+        self.socio_economic = socio_economic
+        self.highest_education = highest_education
+        self.years_of_experience = years_of_experience
+
+        # default values
+        self.others = None
+        self.is_public = False
```

Comment:
I think this should not be a boolean option.
I would say this could be an enum to represent different visibility levels. E.g.:
- Not visible to anyone
- Visible to BIT only
- Visible to Organizations only
- Visible to all users (probably no one will select this though 🤔 )